src/extra.ml

```diff
@@ -212,10 +219,15 @@ module Array =
         let f x y = if not (f x y) then raise Done in
         try iter2 f a1 a2; true with Done -> false
 
-    (** [pp pp_e sep oc a] prints the array list [a] on the channel [oc] using
-        [sep] as separator, and [pp_e] for printing the elements. *)
+    (** [pp pp_elt sep oc a] prints the array list [a] on the channel [oc]
+        using [sep] as separator, and [pp_e] for printing the elements. *)
     let pp : 'a pp -> string -> 'a array pp = fun pp_elt sep oc a ->
-      List.pp pp_elt sep oc (to_list a)
+      let n = Array.length a in
```

Comment:
I think the following would be slightly more readable (and one line shorter):
```ocaml
let n = Array.length a in
if n > 0 then pp_elt oc a.(0);
for i = 1 to n - 1 do
  Format.fprintf oc "%s%a" sep pp_elt a.(i)
done
```

Reply:
But for the record, I think that using `List.pp` was fine. The complexity of the function was the same, and it's not like input/output functions are critical for efficiency here.